pkg/expr/commands.go

```diff
@@ -152,18 +153,27 @@ func (gr *ReduceCommand) NeedsVars() []string {
 
 // Execute runs the command and returns the results or an error if the command
 // failed to execute.
-func (gr *ReduceCommand) Execute(ctx context.Context, vars mathexp.Vars) (mathexp.Results, error) {
+func (gr *ReduceCommand) Execute(_ context.Context, vars mathexp.Vars) (mathexp.Results, error) {
 	newRes := mathexp.Results{}
 	for _, val := range vars[gr.VarToReduce].Values {
-		series, ok := val.(mathexp.Series)
-		if !ok {
+		switch v := val.(type) {
+		case mathexp.Series:
+			num, err := v.Reduce(gr.refID, gr.Reducer, gr.seriesMapper)
+			if err != nil {
+				return newRes, err
+			}
+			newRes.Values = append(newRes.Values, num)
+		case mathexp.Number: // if incoming vars is just a number, any reduce op is just a noop, add it as it is
+			copyV := mathexp.NewNumber(gr.refID, v.GetLabels())
+			copyV.SetValue(v.GetFloat64Value())
+			copyV.AddNotice(data.Notice{
+				Severity: data.NoticeSeverityWarning,
+				Text:     fmt.Sprintf("Query or expression %s produced result of type 'single number'. Reduce operation is not needed.", gr.VarToReduce),
```

Comment:
Maybe "Reduce operation is not needed. Input query or expression %s is already reduced data."

"